Packages/com.unity.inputsystem/InputSystem/API/Input.cs

```diff
@@ -239,62 +268,115 @@ public static class Input
     {
 // RE-ENABLE ME WHEN YOU GONNA IMPLEMENT ME
 #if false
-        // These device collections use "stable" indexing. If a device that was connected disconnects, the
-        // array index that it occupied remains empty until the same device or another device of the same type
-        // connects. This is to enable applications to abstract the concept of a device index and work at a
-        // higher level of "slot" index. This is in contrast to some of the the lower level device collections,
-        // Gamepads.all for example, where the collection is compacted when a device disconnects, and new devices,
-        // and even reconnecting devices, are added to the end of the collection.
+
 
         /// <summary>
-        /// A collection of all keyboards currently connected to the system.
+        /// A collection of all joysticks currently connected to the system.
         /// </summary>
-        public static IReadOnlyList<Keyboard> keyboards { get; }
+        public static IReadOnlyList<Joystick> joysticks { get; }
+#endif
+
+        static Input()
+        {
+            s_GamepadSlotEnums = new ReadOnlyArray<GamepadSlot>(new[]
```

Comment:
Is it a super hardcode version because `Enum.GetValues` allocates or something?

Reply:
Yep. And because it's the simplest possible version of it. It's just nicer than filtering All (and Any was there at the time I wrote it too).